frontend/src/App.tsx

```diff
@@ -1,66 +1,109 @@
 import { useEffect, useState } from "react";
-import { Wallet } from "fuels";
 import "./App.css";
 // Import the contract factory -- you can find the name in index.ts.
 // You can also do command + space and the compiler will suggest the correct name.
 import { CounterContractAbi__factory } from "./contracts";
-
+ 
 // The address of the contract deployed the Fuel testnet
 const CONTRACT_ID =
-  "0x2c38161e42ce14abdaa7950c04b13aa340eb67cf196c94b971d5f0175417f4f4";
-
-//the private key from createWallet.js
-const WALLET_SECRET =
-  "0x07960a5124cd6e52b595aed727d640aed2e32cc587a34c09f1b48615378df752";
-
-// Create a Wallet from given secretKey in this case
-// The one we configured at the chainConfig.json
-const wallet = Wallet.fromPrivateKey(
-  WALLET_SECRET,
-  "https://beta-3.fuel.network/graphql"
-);
-
-// Connects out Contract instance to the deployed contract
-// address using the given wallet.
-const contract = CounterContractAbi__factory.connect(CONTRACT_ID, wallet);
-
+  "0xa0f2b97f32300d7360b5632e682c1508b326da9a00fbe755646d3809aec90672";
+ 
 function App() {
-  const [counter, setCounter] = useState(0);
-  const [loading, setLoading] = useState(false);
-
+  const [connected, setConnected] = useState<boolean>(false);
+  const [account, setAccount] = useState<string>("");
+  const [counter, setCounter] = useState<number>(0);
+  const [loaded, setLoaded] = useState(false);
+ 
   useEffect(() => {
-    async function main() {
-      // Executes the counter function to query the current contract state
-      // the `.get()` is read-only, because of this it don't expand coins.
-      const { value } = await contract.functions.count().get();
-      setCounter(Number(value));
+    setTimeout(() => {
+      checkConnection();
+      setLoaded(true);
+    }, 200)
+    if (connected) getCount();
+  }, [connected, getCount])
+ 
+  async function connect() {
+    if (window.fuel) {
+      try {
+        await window.fuel.connect();
+        const [account] = await window.fuel.accounts();
+        setAccount(account);
+        setConnected(true);
+      } catch (err) {
+        console.log("error connecting: ", err);
+      }
     }
-    main();
-  }, []);
-
+  }
+ 
+  async function checkConnection() {
+    if (window.fuel) {
+      const isConnected = await window.fuel.isConnected();
+      if (isConnected) {
+        const [account] = await window.fuel.accounts();
+        setAccount(account);
+        setConnected(true);
+      }
+    }
+  }
+ 
+  async function getCount() {
+    if (window.fuel) {
+      const wallet = await window.fuel.getWallet(account);
+      const contract = CounterContractAbi__factory.connect(CONTRACT_ID, wallet);
+      const { value } = await contract.functions.count().simulate();
+      setCounter(value.toNumber());
+    }
+  }
```

Comment:
```suggestion
  const getCount = useCallback(async () => {
    if (window.fuel) {
      const wallet = await window.fuel.getWallet(account);
      const contract = CounterContractAbi__factory.connect(CONTRACT_ID, wallet);
      const { value } = await contract.functions.count().simulate();
      setCounter(value.toNumber());
    }
  }, [account]);
```

There was a linter warning about this. The other functions here should likely also be wrapped in a `useCallback`